### PR TITLE
Remove last DocumentMapper reference from MappingLookup

### PR DESCRIPTION
**Pull request by javanna**
As part of #66295 we made QueryShardContext perform mapping lookups through MappingLookup rather than MapperService. That helps as MapperService relies on DocumentMapper which may change througout the execution of the search request. At search time, the percolate query also needs to parse documents, which made us add a parse method to MappingLookup.Such parse method currently relies on calling DocumentMapper#parseDocument through a function, but we would like to rather make this easier to follow. (see https://github.com/elastic/elasticsearch/pull/66295/files#r544639868)

We recently removed the need to provide the entire DocumentMapper to DocumentParser#parse, opening the possibility for using DocumentParser directly when needing to parse a document at query time. This commit adds everything that is needed (namely Mapping, IndexSettings and IndexAnalyzers) to MappingLookup so that it can parse a document through DocumentParser without relying on DocumentMapper.

As a bonus, given that MappingLookup holds a reference to these three additional objects, we can make DocumentMapper rely on MappingLookup to retrieve those and not hold its own same references to them.
Along the same lines, given that MappingLookup holds all that's necessary to parse a document, the signature of DocumentParser#parse can be simplified by replacing most of its arguments with MappingLookup and retrieving what is needed from it.

### CODE REVIEW

File: test/framework/src/main/java/org/elasticsearch/index/mapper/MappingLookupUtils.java [hunk 1]
```diff
@@ -1,36 +0,0 @@
...
-public class MappingLookupUtils {
-    public static MappingLookup fromTypes(MappedFieldType... types) {
-        return fromTypes(Arrays.asList(types), List.of());
-    }
-
-    public static MappingLookup fromTypes(List<MappedFieldType> concreteFields, List<RuntimeFieldType> runtimeFields) {
-        List<FieldMapper> mappers = concreteFields.stream().map(MockFieldMapper::new).collect(toList());
-        return new MappingLookup("_doc", mappers, List.of(), List.of(), runtimeFields, 0, souceToParse -> null, true);
-    }
-}
```
Comment (romseygeek) [inline test/framework/src/main/java/org/elasticsearch/index/mapper/MappingLookupUtils.java:36]: I wonder if it's worth keeping these?  You have quite a few places where you're doing the 'instantiate a RootObjectMapper and build a Mapping' dance.

**Review by romseygeek — CHANGES_REQUESTED** [review]
This is a really nice change, thanks @javanna.  I think it's worth keeping the test utility to build a MappingLookup around, as we end up having lots of similar builder methods on tests without it.

File: test/framework/src/main/java/org/elasticsearch/index/mapper/MappingLookupUtils.java (hunk 1 above)
Comment (javanna) [reply test/framework/src/main/java/org/elasticsearch/index/mapper/MappingLookupUtils.java:36]: I see your point, though I'd try to remove this utility class if possible. I pushed an update that removes the need for the dance most of the times.

File: server/src/test/java/org/elasticsearch/indices/IndicesRequestCacheTests.java [hunk 2]
```diff
@@ -629,7 +631,11 @@ public boolean isFragment() {
         }
     }
 
-    private class TestEntity extends AbstractIndexShardCacheEntity {
+    private static MappingLookup createMappingLookup() {
+        return new MappingLookup(Mapping.EMPTY, emptyList(), emptyList(), emptyList(), null, null, null);
+    }
```
Comment (romseygeek) [inline server/src/test/java/org/elasticsearch/indices/IndicesRequestCacheTests.java:636]: I think you can use `MappingLookup.EMPTY` here?

**Review by romseygeek — APPROVED** [review]
LGTM, thanks @javanna

File: server/src/test/java/org/elasticsearch/indices/IndicesRequestCacheTests.java (hunk 2 above)
Comment (javanna) [reply server/src/test/java/org/elasticsearch/indices/IndicesRequestCacheTests.java:636]: almost :) in some places I do need a different instance, but I no longer need the method I think.

**Review by nik9000 — APPROVED** [review]
LGTM2. Thanks for this!

**Pull request merged**